docs/examples/Database/Database.ipynb

```diff
@@ -0,0 +1,216 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Databases\n",
+    "\n",
+    "This is an example notebook to briefly discuss the use of SQLite3 databases in QCoDeS. Generally, databases are used to store values of `parameters` obtained during a `measurement` in `DataSet` ojects. Specically, storage occurs as part of  the `add_result` method of the `datasaver` object in the context manager. `DataSets` can also be read from the database for analysis.  \n",
+    "\n",
+    "## Imports\n",
+    "\n",
+    "For our example we require only a few modules.\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Logging hadn't been started.\n",
+      "Activating auto-logging. Current session state plus future input saved.\n",
+      "Filename       : C:\\Users\\a-davidpoole\\.qcodes\\logs\\command_history.log\n",
+      "Mode           : append\n",
+      "Output logging : True\n",
+      "Raw input log  : False\n",
+      "Timestamping   : True\n",
+      "State          : active\n",
+      "Qcodes Logfile : C:\\Users\\a-davidpoole\\.qcodes\\logs\\211129-5788-qcodes.log\n"
+     ]
+    }
+   ],
+   "source": [
+    "import qcodes as qc\n",
+    "from qcodes import (\n",
+    "    initialise_database,\n",
+    "    initialise_or_create_database_at,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Initialization, creation and upgrade\n",
+    "\n",
+    "A database can be initialized with a simple function call, `initialize_database`, which will initialize a database located at `~/experiments.db`. This function produces does not return or print a value, but the location of the currently initialized database is stored in the QCoDeS configuration `qc.config.core.db_location`.\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "~/experiments.db\n"
+     ]
+    }
+   ],
+   "source": [
+    "initialise_database()\n",
+    "\n",
+    "print(qc.config.core.db_location)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "When an existing database is initialized, it is always examined and upgraded to the latest version by a tested automatic process. When this occurs, a more detailed output is produced that details the upgrade process. This upgrade occurs with journaling to ensure data safety, to date there have been no instances of data corruption during database upgrades. \n",
```

Comment:
This notebook should probably explain that 

* `initialise_or_create_database_at` does not permanently set the location, e.g. after a restart the config will be back to what it was
* Explain how to set the default db location in the config file